src/mscorlib/shared/System/Collections/Generic/Dictionary.cs

```diff
@@ -759,6 +759,24 @@ IEnumerator IEnumerable.GetEnumerator()
             return new Enumerator(this, Enumerator.KeyValuePair);
         }
 
+        /// <summary>
+        /// Ensures that the capacity of this dictionary is at least the given minimum value
+        /// </summary>
+        /// <param name="capacity"></param>
+        /// <returns></returns>
+        public int EnsureCapacity(int capacity)
+        {
+            if (capacity < 0)
+                ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.capacity);
+            if (_entries == null)
```

Comment:
Elsewhere this is `if (_buckets == null) Initialize(0);` so I'd be consistent. I'd also move this below the check below just so checks are "up front"

Reply:
If I am understanding correctly you are asking to have condition check of ```buckets == null``` after the ``` if (capacity <= _entries.Length) ``` check.

I cannot do that because ``` _buckets ``` and ``` _entries ``` are empty at the same time and may result in null ptr exception

Reply:
Of course if you move below the check for Length you will need to check for null there.

Moving it down means that if I call `EnsureCapacity(0)` on a fresh dictionary, it does not allocate anything.